app/models/uphold_connection.rb

```diff
@@ -207,6 +215,7 @@ def sync_connection!
     # Set uphold_details to a variable, if uphold_access_parameters is nil
     # we will end up makes N service calls everytime we call uphold_details
     # this is a side effect of the memoization
+    Uphold::Refresher.build.call(uphold_connection: self)
```

Comment:
Please only call this if the access token is expired.

Reply:
Telling if it is expired is not straightforward as we only get back the number of seconds the access token is good for (3600). But we don't store the last time that we updated this field, so the comparison would involve adding a new column. Since this is not expected to be called very often, I think it's fine to leave it as is.

Reply:
I'll have to add a new key to the JSON which we generate ourselves, that is, the expiration time.